### PR TITLE
Allow dynamic root tag for FormulateForm

### PR DESCRIPTION
**Pull request by EO2875**
# Changes
We expose a `tag` prop to the `FormulateForm` component:
```xml
<FormulateForm tag="tr" >
    Hello World
</FormulateForm>
```

and it will render as:
```html
<tr>
    Hello World
</tr>
```

(The tag is `form` by default.)

# Rationale

While the `form` tag was useful in vanilla HTML, JavaScript allows us to explore other kinds of inputs.

Our customers must log a large amount of data quickly. We strive to immitate Excel behavior, but with richer and more dynamic UI. We thus use this dynamic tag functionality to create a form-table as follows:

```xml
  <table>
    <tbody>
      <FormulateForm
        v-for="(item, index) in data"
        :key="index"
        tag="tr"
        v-model="data[index]"
        :errors="item.fieldErrors"
      >
        <td>
          <FormulateInput
            type="text"
            name="name"
          />
        </td>
        <td>
          <FormulateInput
            type="number"
            name="quantity"
          />
        </td>
        <td>
          <FormulateErrors
            :context="({ errors: item.formErrors, })"
          />
        </td>
      </FormulateForm>
    </tbody>
  </table>
```

# Disadvantages
However, we understand that using tags other than `form` will lose the semantic meaning. Further investigation into accessibility is required. Perhaps we could use [Vue 3 Teleport](https://v3.vuejs.org/guide/teleport.html) in the future?

# Further Reading
[Form inside a table](https://stackoverflow.com/questions/5967564/form-inside-a-table)
[`form` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefform)

### CODE REVIEW

**Comment by justin-schroeder** [comment]
@EO2875 I'm curious what features of `FormulateForm` you want to use/leverage that aren't available just using standard inputs and what not.

**Comment by EO2875** [comment]
These features are of course available using standard inputs. However, without 
a central `FormulateForm`, we must bind the `value` and `errors` props and each 
`v-on` handler to each individual `FormulateInput`:

```xml
<tr
    v-for="(item, index) in data"
    :key="index"
>
    <td>
        <FormulateInput
            type="text"
            name="name"
            v-model="data[index].name"
            v-bind:errors="item.errors.name"
            v-bind:show-errors="showErrorsBool"
            v-on:input="onInput()"
            v-on:validation="onValidation()"
            v-on:error-visibility="onErrorVisibility()"
        />
    </td>
    <td>
        <FormulateInput
            type="number"
            name="quantity"
            v-model="data[index].quantity"
            v-bind:errors="item.errors.quantity"
            v-bind:show-errors="showErrorsBool"
            v-on:validation="onValidation()"
            v-on:error-visibility="onErrorVisibility()"
        />
    </td>
    <td>
        <FormulateErrors
            :context="({ errors: item.formErrors, })"
        />
    </td>
</tr>
```

I have no doubt that this approach will achieve the desired result in 100% of 
the cases. However, we are left with much repeated code.


Vicente Ruso suggested wrapping the whole table in a `FormulateForm` instead.

```xml
<FormulateForm
    v-model="data"
    errors="errors"
>
    ...
        <tr v-for="(item, index) in data" :key="index">
            <td>
                <FormulateInput :name="`${index}.quantity`" />
            </td>
            ...
        </tr>
</FormulateForm>
```

However, instead of forming this:

```js
data = {
    42: {
        quantity: 56
    },
    ...
}
```

it instead forms:
```js
data = {
    "42.quantity": 56,
    ...
}
```

and does not keep track of nested errors.

In a nutshell, it allows `FormulateForm` to be a table row. The table will 
represent an array of objects, each row-form will be an object instance,
and each cell-input will be a field in each object instance. We can achieve
this using only `FormulateInput`s, but it becomes verbose. 

If you think it is unnecessary and that the `FormulateInput`'s added
verbosity is manageable, I'd understand.
